tests/scripts/arm32_ci_script.sh

```diff
@@ -291,7 +291,7 @@ function cross_build_coreclr_with_docker {
 
     __extraArgs=""
     if [[ "$__buildArch" == "armel" && "$__linuxCodeName" == "tizen" ]]; then
-        __extraArgs="cmakeargs -DFEATURE_GDBJIT=TRUE -PortableBuild=false"
+        __extraArgs="cmakeargs -DFEATURE_NGEN_RELOCS_OPTIMIZATIONS=true cmakeargs -DFEATURE_GDBJIT=true cmakeargs -DCLR_ADDITIONAL_LINKER_FLAGS=\"-Wl,-z,relro\" cmakeargs -DFEATURE_PREJIT=true -PortableBuild=false"
```

Comment:
Do you really need to explicitly enable `relro`? I believe that it has been enabled by default in the compilers/linkers for many years.

Reply:
I think it's better to enable `relro` explicilty, because it isn't enabled by default in some cases. For example clang enables `relro` by default only for openSUSE, ubuntu, alpine and android:
https://github.com/llvm/llvm-project/blob/81f433b48c1892034e7061452ffb158e5244f62d/clang/lib/Driver/ToolChains/Linux.cpp#L250
Also `ld` doesn't enable it by default on ubuntu 16.04:
```
root@b73b83ef9669:/# cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.4 LTS"
root@b73b83ef9669:/# clang-6.0 /tmp/1.c -v
clang version 6.0.0-1ubuntu2~16.04.1 (tags/RELEASE_600/final)
...
 "/usr/bin/ld" -z relro --hash-style=gnu --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crt1.o /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crti.o /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/crtbegin.o -L/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0 -L/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../.. -L/usr/lib/llvm-6.0/bin/../lib -L/lib -L/usr/lib /tmp/1-316406.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/crtend.o /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crtn.o
root@b73b83ef9669:/# ./checksec --file a.out
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH	Symbols		FORTIFY	Fortified	Fortifiable  FILE
Partial RELRO   No canary found   NX enabled    No PIE          No RPATH   No RUNPATH   66 Symbols     No	0		1	a.out

root@b73b83ef9669:/# sed 's/xenial/1/g' -i /etc/lsb-release
root@b73b83ef9669:/# cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=1
DISTRIB_DESCRIPTION="Ubuntu 16.04.4 LTS"
root@b73b83ef9669:/# clang-6.0 /tmp/1.c -v
clang version 6.0.0-1ubuntu2~16.04.1 (tags/RELEASE_600/final)
...
 "/usr/bin/ld" --hash-style=both --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crt1.o /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crti.o /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/crtbegin.o -L/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0 -L/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../.. -L/usr/lib/llvm-6.0/bin/../lib -L/lib -L/usr/lib /tmp/1-057dee.o -lgcc --as-needed -lgcc_s --no-as-needed -lc -lgcc --as-needed -lgcc_s --no-as-needed /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/crtend.o /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.0/../../../x86_64-linux-gnu/crtn.o
root@b73b83ef9669:/# ./checksec --file a.out
RELRO           STACK CANARY      NX            PIE             RPATH      RUNPATH	Symbols		FORTIFY	Fortified	Fortifiable  FILE
No RELRO        No canary found   NX enabled    No PIE          No RPATH   No RUNPATH   67 Symbols     No	0		1	a.out
```
But as we usually use ubuntu it isn't really needed and I can remove it.

Reply:
@janvorli Should we enable `relro` in some central location?

Reply:
Yes, we should. Even our official builds (at least for x64 linux) don't have the relro enabled.

Reply:
Can we enable `relro` in `configurecompiler.cmake`?
```diff
diff --git a/configurecompiler.cmake b/configurecompiler.cmake
index 41da56307b..f5ce3873de 100644
--- a/configurecompiler.cmake
+++ b/configurecompiler.cmake
@@ -350,8 +350,8 @@ endif(CLR_CMAKE_PLATFORM_UNIX)
 
 if(CLR_CMAKE_PLATFORM_LINUX)
   set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,--noexecstack")
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1")
+  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1 -Wl,-z,relro")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1 -Wl,-z,relro")
 endif(CLR_CMAKE_PLATFORM_LINUX)
 if(CLR_CMAKE_PLATFORM_FREEBSD)
   set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,--noexecstack")
```

Reply:
Yes, that's the right place to do that.

Reply:
Thank you! I've removed setting additional linker flags here and created new PR https://github.com/dotnet/coreclr/pull/25473